### PR TITLE
feature: make `rustls` default in `fluvio`, move to `reqwest` from `surf`

### PR DESCRIPTION
**Pull request by ozgrakkurt**
- Made `rustls` feature on `fluvio` the default
- Made most components use `rustls` if it is enabled. It used to use `openssl` anyway because of a compatibility problem.
-  Removed usage of `surf` because it is not maintained and added `reqwest` instead.

closes https://github.com/infinyon/fluvio/issues/3091
closes https://github.com/infinyon/fluvio/issues/3113
closes https://github.com/infinyon/fluvio/issues/3070

Seems like a bunch of things need to be manually checked using the cli before we are sure this works?. One problem might be that we need `async-std/tokio1` enabled and I enabled it in `fluvio` crate but this will be problem if the sub-crates are used somewhere else where `async-std/tokio1` isn't enabled.

### CODE REVIEW

**Comment by ozgrakkurt** [comment]
@digikata it isn't possible to get rid of `openssl` entirely for now because we depend on `cargo-generate` which doesn't seem to give an option to use `rustls` or run without tls.

File: Cargo.toml [hunk 1]
```diff
@@ -97,13 +97,31 @@ colored = "2.0.0"
...
 which = "4.1.0"
+reqwest = { version = "0.11", default-features = false, features = ["json", "gzip"] }
 
 # fluvio dep
 k8-client = { version = "10.0.0" }
 k8-types = { version = "0.8.0" }
 k8-config = { version = "2.0.0"}
 k8-metadata-client = { version = "5.1.0" }
-fluvio-future = { version = "0.4.0" }
+fluvio-future = { version = "0.4", default-features = false, features = [
```
Comment (sehz) [inline Cargo.toml:107]: don't add features at workspace dependency

**Review by sehz — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (ozgrakkurt) [reply Cargo.toml]: done

File: Cargo.toml [hunk 2]
```diff
@@ -97,13 +97,31 @@ colored = "2.0.0"
 semver = "1.0.3"
 url = "2.1.1"
 which = "4.1.0"
+reqwest = { version = "0.11", default-features = false, features = ["json", "gzip"] }
```
[the remark this answers is not in the document]
Comment (ozgrakkurt) [reply Cargo.toml]: done

**Comment by ozgrakkurt** [comment]
@sehz moved features from workspace to individual crates, made everything use `rustls` as much as I can and the rest uses `openssl`/`native-tls`.

removed tls features on `fluvio` crate.

**Comment by ozgrakkurt** [comment]
seems like there are errors about doing `fluvio-future = { workspace = true }`, how should I handle these? Do we need the version to be specified in each of these crates?

**Comment by sehz** [comment]
you don't need to specify version but need specific feature enabled in each crate that need optional features

**Comment by ozgrakkurt** [comment]
@sehz bumped versions also fixed some mistakes. Also this is a breaking change since I removed the `openssl` and `rustls` features from `fluvio` crate

**Review by digikata — COMMENTED** [review]
Looks really good so far, after it settles down more I wonder if it might be worth splitting apart the PR to some boundaries to phase it in. It's a large number of changes, so separating possible side effects to smaller sets of changes makes it easier on downstream projects like fluvio-cloud-ops.  e.g. the fluvio-future syncup to a workspace dep might be worth peeling off and merging to master before this.

**Comment by ozgrakkurt** [comment]
> Looks really good so far, after it settles down more I wonder if it might be worth splitting apart the PR to some boundaries to phase it in. It's a large number of changes, so separating possible side effects to smaller sets of changes makes it easier on downstream projects like fluvio-cloud-ops. e.g. the fluvio-future syncup to a workspace dep might be worth peeling off and merging to master before this.

Makes sense, probably gonna have to do this like `reqwest` change and then another PR for the TLS. My bad on this PR

**Comment by digikata** [comment]
> Makes sense, probably gonna have to do this like `reqwest` change and then another PR for the TLS. My bad on this PR

No worries, the exploratory work had to be done to find out what the extent of the changes might be.

**Pull request closed without merging**